test/views/payments/WhatYouOweViewSpec.scala

```diff
@@ -28,255 +28,291 @@ class WhatYouOweViewSpec extends ViewBaseSpec {
+          "has the correct link text" in {
+            elementText("p.govuk-body:nth-of-type(6) > a") shouldBe "make a payment (opens in a new tab)"
+          }
+
+          "has the correct link location" in {
+            element("p.govuk-body:nth-of-type(6) > a").attr("href") shouldBe mockConfig.unauthenticatedPaymentsUrl
+          }
+        }
       }
     }
 
-    "have a section for guidance if what the user owes is missing" which {
+    "there is at least one overdue payment and the overdueTimeToPayDescriptionEnabled feature switch is on" should {
 
-      "has the correct heading" in {
-        elementText("h2:nth-of-type(2)") shouldBe "What you owe is incorrect or missing"
-      }
+      lazy val view = whatYouOweView(whatYouOweViewModel2Charge, Html(""))
+      lazy implicit val document: Document = Jsoup.parse(view.body)
 
-      "has the correct first paragraph" which {
+      "not have a section for guidance if the user cannot pay today" in {
 
-        "has the correct text" in {
-          elementText("p.govuk-body:nth-of-type(5)") shouldBe "If what you owe is incorrect, check if you can " +
-            "correct errors on your VAT Return (opens in a new tab)."
-        }
+        mockConfig.features.overdueTimeToPayDescriptionEnabled(false)
+        elementExtinct("#cannot-pay-heading")
+        elementExtinct("#cannot-pay-paragraph")
+      }
+    }
 
-        "has the correct link text" in {
-          elementText("p.govuk-body:nth-of-type(5) > a") shouldBe "correct errors on your VAT Return (opens in a new tab)"
-        }
+    "there are no overdue payments" should {
 
-        "has the correct link location" in {
-          element("p.govuk-body:nth-of-type(5) > a").attr("href") shouldBe mockConfig.govUKCorrections
-        }
-      }
+      lazy val view = whatYouOweView(whatYouOweViewModel, Html(""))
+      lazy implicit val document: Document = Jsoup.parse(view.body)
 
-      "has the correct second paragraph" which {
+      "not have a section for guidance if the user cannot pay today" when {
 
-        "has the correct text" in {
-          elementText("p.govuk-body:nth-of-type(6)") shouldBe "After you have submitted a return, it can take " +
-            "24 hours for what you owe to show here. You can still make a payment (opens in a new tab)."
-        }
+        "the feature switch is on" in {
 
-        "has the correct link text" in {
-          elementText("p.govuk-body:nth-of-type(6) > a") shouldBe "make a payment (opens in a new tab)"
+          mockConfig.features.overdueTimeToPayDescriptionEnabled(true)
+          elementExtinct("#cannot-pay-heading")
+          elementExtinct("#cannot-pay-paragraph")
         }
 
-        "has the correct link location" in {
-          element("p.govuk-body:nth-of-type(6) > a").attr("href") shouldBe mockConfig.unauthenticatedPaymentsUrl
+        "the feature switch is off" in {
```

Comment:
You would have to make another view/document to re-toggle the feature switch and test this scenario. Currently this test will just use the view that was created above, where the switch was set to true; it won't re-render it.